### DISCUSSION TITLE
Simplify the link button

### DISCUSSION THREAD

**Discussion by mikesnoeren** [craftcms/ckeditor#78] [Ideas] [answered]
Currently, the Insert Link button in the CraftCMS CKEditor plugin has two actions: the left action is to insert a custom link, and the right action is to open a dropdown to insert an entry, asset, or category.

Ideally, the button would have only one action that shows a dropdown with all available options. If this is not possible, I suggest adding an "insert a custom link" option to the dropdown so that all options are contained within it.

**Comment by brandonkelly** [comment]
Converted this to a discussion because it’s currently working as intended.

**Comment by alexkuzmin** [comment]
It would be also very nice if the actions of Link button could be enabled/disabled, e.g. to enable only "Link to an entry" and disable other actions.

**Reply by brandonkelly** [reply]
We could certainly add “Available Sections” and “Available Category Groups” field settings, similar to “Available Volumes”. If all sections are deselected, “Link to an entry” wouldn’t show up, for example.

**Comment by brandonkelly** [comment, marked as answer]
We can’t make the entire button act as the menu trigger, because then it wouldn’t be possible to edit existing link URLs.

Added an “Insert link” option to the link dropdown, for the next release, though (4693a7fde843a2f12a21af233ef11f28b2f265b3).

**Reply by brandonkelly** [reply]
3.3.0 is out with that change 🎉
